libsql/internal/ws/driver.go

```diff
@@ -168,15 +181,69 @@ func convertArgs(args []driver.NamedValue) params {
 	return params{PositinalArgs: posArgs, NamedArgs: namedArgs}
 }
 
-func (c *conn) ExecContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Result, error) {
+func (c *conn) ExecContext(
+	ctx context.Context,
+	query string,
+	args []driver.NamedValue,
+) (driver.Result, error) {
+	now := time.Now()
+	age := now.Sub(c.ws.opentime)
+	if age.Milliseconds() > maxConnAge.Milliseconds() {
+		ws, err := connect(c.url, c.jwt)
+		if err != nil {
+			return nil, err
+		}
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		c.ws.conn = ws.conn
+		c.ws.opentime = ws.opentime
+		return execContext(c, ctx, query, args, false)
+	}
+	return execContext(c, ctx, query, args, false)
+}
+
+func (c *conn) QueryContext(
+	ctx context.Context,
+	query string,
+	args []driver.NamedValue,
+) (driver.Rows, error) {
+	now := time.Now()
+	age := now.Sub(c.ws.opentime)
+	if age.Milliseconds() > maxConnAge.Milliseconds() {
+		ws, err := connect(c.url, c.jwt)
+		if err != nil {
+			return nil, err
+		}
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		c.ws.conn = ws.conn
+		c.ws.opentime = ws.opentime
```

Comment:
The reconnection logic is duplicated. Please extract it to a separate function like (c *conn) maybeReconnect()